GPU/GLES/ShaderManager.h

```diff
@@ -110,54 +110,110 @@ class LinkedShader {
 	int u_lightdiffuse[4];  // each light consist of vec4[3]
 	int u_lightspecular[4];  // attenuation
 	int u_lightambient[4];  // attenuation
+
+	int u_tess_pos_tex;
+	int u_tess_tex_tex;
+	int u_tess_col_tex;
+	int u_spline_count_u;
+	int u_spline_count_v;
+	int u_spline_type_u;
+	int u_spline_type_v;
 };
 
 enum {
-	DIRTY_PROJMATRIX = (1 << 0),
-	DIRTY_PROJTHROUGHMATRIX = (1 << 1),
-	DIRTY_FOGCOLOR = (1 << 2),
-	DIRTY_FOGCOEF = (1 << 3),
-	DIRTY_TEXENV = (1 << 4),
-	DIRTY_ALPHACOLORREF = (1 << 5),
+	_DIRTY_PROJMATRIX = 0,
```

Comment:
I don't like these _DIRTY and the defines after. Please use the old format (though 1ULL etc of course), it's readable enough. Alternatively, use FLAG_BIT directly in the enum declaration.